Indentation cleanup, general formatting and variable renaming

Gwenwyfar opened the pull request:
Just whitespace changes. Cheers for mass auto-converting.

Gwenwyfar committed:
Indentation cleanup

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

Gwenwyfar committed:
Formatting

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

Gwenwyfar commented:
Some more general formatting while we're at it.

I believe it doesn't make any difference, but just to note, there were two instances of "FALSE" and "NULL" that got changed to lowercase. Other than that, this is the same as above, only spaces and whitespace changes.

live627 commented:
You   are correct in that they don't make a difference - PHP doesn't care for case in  those keywords.

Sesquipedalian commented on Sources/ManagePermissions.php:
```diff
@@ -163,7 +163,7 @@ function PermissionIndex()
 		FROM {db_prefix}membergroups' . (empty($modSettings['permission_enable_postgroups']) ? '
-		WHERE min_posts = {int:min_posts}' : '') . '
+			WHERE min_posts = {int:min_posts}' : '') . '
 		ORDER BY id_parent = {int:not_inherited} DESC, min_posts, CASE WHEN id_group < {int:newbie_group} THEN id_group ELSE 4 END, group_name',
```
I don't think we normally indent this. We should keep it consistent with the rest.

Sesquipedalian commented on Sources/ManageServer.php:
```diff
@@ -302,7 +302,7 @@ function AlignURLsWithSSLSetting($new_force_ssl = 0)
 		SELECT id_theme, variable, value
 		  FROM {db_prefix}themes
-		 WHERE variable in ({string:themeurl}, {string:imagesurl})
+		WHERE variable in ({string:themeurl}, {string:imagesurl})
```
There's some weird indentation happening in this whole query. It should probably all be fixed, not just this one line.

Gwenwyfar replied:
What would you say we should use as default indentation for queries? I don't normally deal with them and I can't quite get something in my head between the inconsistencies and the different format some editors suggest me. Though I see we always indent AND.

Sesquipedalian replied:
As a general rule, we align the SQL keywords that start a main logical clause of a query (e.g. SELECT, FROM, WHERE, UPDATE, ORDER BY, etc.) each on their own line and using the same indentation. 

The main exception is the JOIN keywords (JOIN, INNER JOIN, LEFT JOIN, etc.) which are always indented one more tab stop. This is because JOIN ultimately functions to expand the number of tables used in the FROM clause, and so we indent it in a way that shows it as subordinate to FROM.

Keywords that function more like inline logic within a clause (e.g. AND, OR, IN, AS, ASC, DESC, etc.) can either be left inline or moved to a new line that is indented another tab stop in order to show that it is subordinate to the clause it is part of.

---

The reason that this particular case is weird is simply that someone decided to right-align the major keywords (SELECT, FROM, WHERE) instead of left-aligning them. Just take out the extra spaces in front of those words, and this one should be good.

Gwenwyfar replied:
Alright. I'll do a second revision for those and the breaks then.

Sesquipedalian commented on Sources/ManageServer.php:
```diff
@@ -323,7 +323,7 @@ function AlignURLsWithSSLSetting($new_force_ssl = 0)
 				UPDATE {db_prefix}themes
 				   SET value = {string:theme_val}
-				 WHERE variable = {string:theme_var}
+				WHERE variable = {string:theme_var}
```
There's some weird indentation happening in this whole query, too. It should probably all be fixed, not just this one line.

Sesquipedalian commented on Sources/Subs-Db-mysql.php:
```diff
@@ -978,18 +978,18 @@ function smf_db_fetch_all($request)
-			);
+		$mysql_error_data_prep = mysqli_prepare($db_connection,
+			'INSERT INTO ' . $db_prefix . 'log_errors(id_member, log_time, ip, url, message, session, error_type, file, line, backtrace)
+			VALUES(		?,		?,		unhex(?), ?, 		?,		?,			?,		?,	?, ?)'
```
Those long runs of whitespace between values probably aren't doing anything useful.

Sesquipedalian commented on Sources/Subs-Db-mysql.php:
```diff
@@ -978,18 +978,18 @@ function smf_db_fetch_all($request)
-													VALUES(		?,		?,		unhex(?), ?, 		?,		?,			?,		?,	?, ?)'
-			);
+		$mysql_error_data_prep = mysqli_prepare($db_connection,
+			'INSERT INTO ' . $db_prefix . 'log_errors(id_member, log_time, ip, url, message, session, error_type, file, line, backtrace)
```
A line break and an extra tab between `log_errors` and the parenthesis would make this much more legible.

Sesquipedalian commented on Sources/Subs-Db-postgresql.php:
```diff
@@ -966,18 +968,18 @@ function smf_db_fetch_all($request)
-													VALUES(		$1,		$2,		$3, $4, 	$5,		$6,			$7,		$8,	$9, $10)'
-			);
+		$pg_error_data_prep = pg_prepare($db_connection, 'smf_log_errors',
+			'INSERT INTO ' . $db_prefix . 'log_errors(id_member, log_time, ip, url, message, session, error_type, file, line, backtrace)
```
Again, a line break and an extra tab between `log_errors` and the parenthesis would make this much more legible.

Sesquipedalian commented on Sources/Subs-Db-postgresql.php:
```diff
@@ -966,18 +968,18 @@ function smf_db_fetch_all($request)
-			);
+		$pg_error_data_prep = pg_prepare($db_connection, 'smf_log_errors',
+			'INSERT INTO ' . $db_prefix . 'log_errors(id_member, log_time, ip, url, message, session, error_type, file, line, backtrace)
+			VALUES(		$1,		$2,		$3, $4, 	$5,		$6,			$7,		$8,	$9, $10)'
```
Again, the whitespace between values can be removed.

Sesquipedalian commented on Sources/Subs.php:
```diff
@@ -199,7 +199,7 @@ function updateStats($type, $parameter1 = null, $parameter2 = null)
 				SELECT SUM(num_topics + unapproved_topics) AS total_topics
 				FROM {db_prefix}boards' . (!empty($modSettings['recycle_enable']) && $modSettings['recycle_board'] > 0 ? '
-				WHERE id_board != {int:recycle_board}' : ''),
+					WHERE id_board != {int:recycle_board}' : ''),
```
We don't normally indent WHERE clauses differently than FROM clauses.

Sesquipedalian commented on Sources/Subs.php:
```diff
@@ -259,9 +259,9 @@ function updateStats($type, $parameter1 = null, $parameter2 = null)
+			ELSE 0
+			END' . ($parameter1 != null ? '
+				WHERE ' . (is_array($parameter1) ? 'id_member IN ({array_int:members})' : 'id_member = {int:members}') : ''),
 				array(
```
We don't normally indent WHERE clauses differently than FROM clauses.

Sesquipedalian commented on Sources/Subs.php:
```diff
@@ -2222,7 +2223,7 @@ function parse_bbc($message, $smileys = true, $cache_id = '', $parse_tags = arra
 						# Preceded by a non-domain character or start of line
-						(?<=^|[^\p{L}\p{M}\p{N}\-\.])
+							(?<=^|[^\p{L}\p{M}\p{N}\-\.])
 
```
Please change this one back.

Sesquipedalian commented:
This PR gives me the warm fuzzies.

Gwenwyfar commented:
> This PR gives me the warm fuzzies.

Is that good or bad? xD

Also, in the terms of getting everything consistent, I felt like half the time we indent break;, and the other half we don't (same level as case), so I left most alone after a while...

frandominguezl commented:
Apparently, good! https://www.urbandictionary.com/define.php?term=warm%20fuzzies

Sesquipedalian commented:
> Also, in the terms of getting everything consistent, I felt like half the time we indent break;, and the other half we don't (same level as case), so I left most alone after a while...

It's no different than anything else in PHP. If it's inside a function or governed by an `if` or `case` statement or whatever, then indent it another tab stop the same way that you would anything else.

Gwenwyfar commented:
Hahaha. Well, then I'll indulge a little more. Last time I did one of those was with Oldiesmann in the template files, so I thought I might drive you a little crazy with it.

What about those? https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/PersonalMessage.php#L2669
There's few, and I'd guess there should be no `;`, but I'm not 100% sure.

Sesquipedalian commented:
You are correct. The semicolon should not be there.

Gwenwyfar committed:
More formatting

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

Gwenwyfar committed:
Remove semicolon after conditionals

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

live627 commented:
Conflicting files
- Sources/Subs.php
- other/upgrade.php

Gwenwyfar committed:
Typos... Typos everywhere!

Run for cover and shoot them down!

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

Gwenwyfar commented:
> Conflicting files
> 
>     * Sources/Subs.php
> 
>     * other/upgrade.php

Conflicts make me a sad panda. I still don't know how to fix them nicely in a PR (and github's interface is not nicely). I'll leave it to you.

Gwenwyfar commented on other/upgrade_1-1.sql:
```diff
@@ -1009,8 +1009,8 @@ if (!$has_attachmentType_column)
 	{
-		$clean_name = strtr($row['filename'], '������������������������������������������������������������', 'SZszYAAAAAACEEEEIIIINOOOOOOUUUUYaaaaaaceeeeiiiinoooooouuuuyy');
-		$clean_name = strtr($clean_name, array('�' => 'TH', '�' => 'th', '�' => 'DH', '�' => 'dh', '�' => 'ss', '�' => 'OE', '�' => 'oe', '�' => 'AE', '�' => 'ae', '�' => 'u'));
+		$clean_name = strtr($row['filename'], '������������������������������������������������������������', 'SZszYAAAAAACEEEEIIIINOOOOOOUUUUYaaaaaaceeeeiiiinoooooouuuuyy');
```
Ugh... Phpstorm, what have you done. Was it the file encoding? :/

Edit: Ok, so it got changed to UTF-8 from Windows format. Didn't even know we had anything around that wasn't UTF-8.

Gwenwyfar replied:
Apparently the conversion itself got borked, rather than the encoding itself, so I left it as UTF-8 and just fixed the characters. If we really want it on windows encoding for some reason I can change it back.

Gwenwyfar committed:
Fix special characters

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

live627 committed:
Merge remote-tracking branch 'SimpleMachines/release-2.1' into indentation

Gwenwyfar committed:
Rename some language variables

Better names and consistency for some language variables.

List of changes:
lfyi -> smf_news_cant_connect
errlog -> errorlog
upgrade_empty_errlog -> upgrade_empty_errorlog
errlog_desc -> errorlog_desc
errlog_no_entries -> errorlog_no_entries
totalTimeLogged2 -> total_time_logged_days
totalTimeLogged3 -> total_time_logged_hours
totalTimeLogged4 -> total_time_logged_minutes
totalTimeLogged5 -> total_time_logged_d
totalTimeLogged6 -> total_time_logged_h
totalTimeLogged7 -> total_time_logged_m
search_thisbrd -> search_thisboard
post_becomesUnapproved -> post_becomes_unapproved
boardsEdit -> boards_edit
bbcTagsToUse_select -> enabled_bbc_select
bbcTagsToUse_select_all -> enabled_bbc_select_all
subAction_fail -> sub_action_fail
catConfirm -> cat_delete_confirm
boardConfirm -> board_delete_confirm
catEdit -> cat_edit
catModify -> cat_modify
repair_stats_topics_1 -> repair_topic_wrong_first_id
repair_stats_topics_2 -> repair_topic_wrong_last_id
repair_stats_topics_3 -> repair_topic_wrong_replies
repair_stats_topics_4 -> repair_topic_wrong_unapproved_number
repair_stats_topics_5 -> repair_topic_wrong_approval
pollMode -> poll_mode

All $editortxt strings were changed into names, rather than have their text copy-pasted into the variable. 

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

Gwenwyfar commented:
I was going to make one commit for each variable type, but there were many more language ones than I expected, so I'm splitting it in further commits. This is about half the language files done.

$.sceditor.locale may need to have the copy/paste strings updated if they are being used as var names, could someone check this? Though it seems we don't use them, but mods might.

Gwenwyfar committed:
Undo poll_mode rename

Settings are too special to rename without doing the entire chain.

Signed-off-by: Gwenwyfar <gwenwyfar@simplemachines.org>

jdarwood007 commented:
@frandominguez03 
Closed issue, but this PR contained $txt changes.  Any issues with getting those to LE?

frandominguezl commented:
We should release RC1 language packs once we officially release 2.1 RC1

Gwenwyfar commented:
@Sesquipedalian did you check with the editor locale strings inside the js file? Or is it all good?

Sesquipedalian commented:
Do you mean the changes you made to the $editortxt variables? If so, yes, and it all works nicely.

Gwenwyfar commented:
Alright then :)

Wasn't sure whether some in the js files were the variable name or duplicate strings, the way they got named, heh.
